src/conduit/api.cljs

```diff
@@ -22,17 +23,34 @@
   (str "https://conduit.productionready.io/api/" path))
 
 (defn- parse-body [res]
-  (-> (:body res)
+  (-> res
       js/JSON.parse
       (js->clj :keywordize-keys true)))
 
+(defn- ->xhr [uri xhr-fn params]
+  (-> uri
+      (xhr-fn params)
+      (p/then (fn [{status :status body :body :as response}]
+                (condp = status
+                  status/ok (p/resolved (parse-body body))
+                  (p/rejected (parse-body body)))))))
+
 (defn fetch
   ([endpoint]
    (fetch endpoint nil))
   ([endpoint params]
    (fetch endpoint params nil))
   ([endpoint params slug]
-   (-> (->endpoint endpoint slug)
-       ->uri
-       (xhr/get {:query-params params})
-       (p/then parse-body))))
+   (fetch endpoint params slug nil))
+  ([endpoint params slug method]
+   (fetch endpoint params slug method nil))
+  ([endpoint params slug method type]
+   (fetch endpoint params slug method type nil))
+  ([endpoint params slug method type headers]
+   (let [xhr-fn (if (= method :post) xhr/post xhr/get)
+         xhr-params {:query-params (when (not (= method :post)) params)
+                     :body (when (= method :post) (.stringify js/JSON (clj->js params)))
+                     :headers (if (= type :json) (merge headers {"Content-Type" "application/json"}) headers)}]
```

Comment:
Maybe change this into `case` form, so it would be extensible for future request type options